c_glib/arrow-glib/table.cpp

```diff
@@ -553,9 +553,9 @@ garrow_table_concatenate(GArrowTable *table,
     arrow_tables.push_back(arrow_other_table);
   }
   std::shared_ptr<arrow::Table> arrow_concatenated_table;
-  auto status = arrow::ConcatenateTables(arrow_tables, &arrow_concatenated_table);
-  if (garrow_error_check(error, status, "[table][concatenate]")) {
-    return garrow_table_new_raw(&arrow_concatenated_table);
+  auto arrow_concatenated_table = arrow::ConcatenateTables(arrow_tables);
+  if (garrow::check(error, arrow_concatenated_table, "[table][concatenate]")) {
```

Comment:
This looks like an opportunity for a `Result` helper function:
```suggestion
  if (auto t = garrow::check_moved(error, std::move(arrow_concatenated_table), "[table][concatenate]")) {
    return garrow_table_new_raw(&t);
```

Reply:
That's interesting but I think that this is out of scope of this pull request.
We can consider this approach in a separated issue.

Cc: @shiro615 @mrkn

Reply:
It's very interesting to me, too.  I've create [the new ticket](https://issues.apache.org/jira/browse/ARROW-7504) to do it.

Reply:
Mark as resolved (out of scope).